monai/bundle/scripts.py

```diff
@@ -1539,3 +1521,61 @@ def init_bundle(
         copyfile(str(ckpt_file), str(models_dir / "model.pt"))
     elif network is not None:
         save_state(network, str(models_dir / "model.pt"))
+
+
+def create_workflow(
+    workflow_name: str | BundleWorkflow | None = None,
+    config_file: str | Sequence[str] | None = None,
+    args_file: str | None = None,
+    **kwargs: Any,
+) -> Any:
+    """
+    Specify `bundle workflow` to create monai bundle workflows.
+    The workflow should be subclass of `BundleWorkflow` and be available to import.
+    It can be MONAI existing bundle workflows or user customized workflows.
+
+    Typical usage examples:
+
+    .. code-block:: python
+
+        # Specify config_file path to create workflow:
+        workflow = create_workflow(config_file="/workspace/spleen_ct_segmentation/configs/train.json", workflow="train")
+
+        # Set the workflow to other customized BundleWorkflow subclass to create workflow:
+        workflow = create_workflow(workflow_name=CustomizedWorkflow)
+
+    Args:
+        workflow_name: specified bundle workflow name, should be a string or class, default to "ConfigWorkflow".
+        config_file: filepath of the config file, if it is a list of file paths, the content of them will be merged.
+        args_file: a JSON or YAML file to provide default values for this API.
+            so that the command line inputs can be simplified.
+        kwargs: arguments to instantiate the workflow class.
+
+    """
+    _args = _update_args(args=args_file, workflow_name=workflow_name, config_file=config_file, **kwargs)
+    _log_input_summary(tag="run", args=_args)
+    (workflow_name, config_file) = _pop_args(
+        _args, workflow_name=ConfigWorkflow, config_file=None
+    )  # the default workflow name is "ConfigWorkflow"
+    if isinstance(workflow_name, str):
+        workflow_class, has_built_in = optional_import("monai.bundle", name=str(workflow_name))  # search built-in
+        if not has_built_in:
+            workflow_class = locate(str(workflow_name))  # search dotted path
+        if workflow_class is None:
+            raise ValueError(f"cannot locate specified workflow class: {workflow_name}.")
+    elif issubclass(workflow_name, BundleWorkflow):  # type: ignore
+        workflow_class = workflow_name
+    else:
+        raise ValueError(
+            "Argument `workflow_name` must be a bundle workflow class name"
+            f"or subclass of BundleWorkflow, got: {workflow_name}."
+        )
+
+    if config_file is not None:
+        workflow_ = workflow_class(config_file=config_file, **_args)
+    else:
+        workflow_ = workflow_class(**_args)
+
+    workflow_.initialize()
```

Comment:
Hi @Nic-Ma, In most cases I've seen, workflow always comes after initialize. For example, if I want to get the property, I can skip this step; I'm not sure if including it would cause issues in any way.

Reply:
If no obvious requirement, we should try to decouple the logic. I would suggest to only create the workflow here.
@wyli What do you think?

Thanks.

Reply:
it seems `initialize()` is needed in almost all the use cases https://github.com/Project-MONAI/MONAI/blob/6f5005fddfaf935cc9b723be823f446a09136d11/monai/bundle/workflows.py#L316-L317

I think we can hard-code this and we can make it optional if we need the flexibility.

Reply:
OK, sure.